tracer/src/Datadog.Trace/Ci/EventModel/CIVisibilityEventsFactory.cs

```diff
@@ -0,0 +1,37 @@
+﻿// <copyright file="CIVisibilityEventsFactory.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+namespace Datadog.Trace.Ci.EventModel;
+
+internal static class CIVisibilityEventsFactory
+{
+    public static IEvent FromSpan(Span span)
+    {
+        switch (span.Type)
+        {
+            case SpanTypes.Test:
+                return new TestEvent(span);
+            case SpanTypes.TestSuite:
+                return new TestSuiteEvent(span);
+            case SpanTypes.TestModule:
+                return new TestModuleEvent(span);
+            case SpanTypes.TestSession:
+                return new TestSessionEvent(span);
+        }
+
+        return new SpanEvent(span);
+    }
```

Comment:
I'm going to keep forcing switch expressions on people whenever I can 😛 
```suggestion
    public static IEvent FromSpan(Span span)
        => span.Type switch
        {
            SpanTypes.Test => new TestEvent(span),
            SpanTypes.TestSuite => new TestSuiteEvent(span),
            SpanTypes.TestModule => new TestModuleEvent(span),
            SpanTypes.TestSession => new TestSessionEvent(span),
            _ => new SpanEvent(span),
        };
```

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3276/commits/cbd9fb841afaed7a6ae9fbb904b4aa2734341f83